rero_mef/agents/api.py

```diff
@@ -0,0 +1,249 @@
+            mef_record = MefRecord.get_mef_by_viaf_pid(viaf_record.pid)
+        else:
+            mef_record = MefRecord.get_mef_by_agent_pid(self.pid, self.agent)
+        if self.deleted:
+            if mef_record:
+                mef_action = Action.DELETEAGENT
+                mef_record.pop(self.agent, None)
+                mef_record = mef_record.update(
+                    data=mef_record,
+                    dbcommit=dbcommit,
+                    reindex=reindex
+                )
+            else:
+                mef_action = Action.DISCARD
+        else:
+            if mef_record:
+                mef_action = Action.UPDATE
+                mef_record = mef_record.update(
+                    data=mef_data,
+                    dbcommit=dbcommit,
+                    reindex=reindex
+                )
+            else:
+                mef_action = Action.CREATE
+                mef_record = MefRecord.create(
+                    data=mef_data,
+                    dbcommit=dbcommit,
+                    reindex=reindex,
+                )
+        if reindex:
+            MefRecord.update_indexes()
+        return mef_record, mef_action, viaf_record, got_online
+
+    @classmethod
+    def create_or_update_agent_mef_viaf(cls, data, id_=None, delete_pid=True,
+                                        dbcommit=False, reindex=False,
+                                        test_md5=False, online=True,
+                                        verbose=False):
+        """Create or update agent, Mef and Viaf record."""
+        try:
+            persistent_id = PersistentIdentifier.query.filter_by(
+                pid_type=cls.provider.pid_type,
+                pid_value=data.get('pid')
+            ).one()
+            if persistent_id.status == PIDStatus.DELETED:
+                return None, Action.ALREADYDELETED, None, Action.DISCARD, \
+                    None, False
+        except Exception as err:
+            pass
+
+        record, action = cls.create_or_update(
+            data=data,
+            id_=id_,
+            delete_pid=delete_pid,
+            dbcommit=dbcommit,
+            reindex=reindex,
+            test_md5=test_md5
+        )
+        if record.deleted:
+            mef_record, mef_action = record.delete_from_mef(
+                dbcommit=dbcommit,
+                reindex=reindex,
+                verbose=verbose
+            )
+            record.delete(
+                dbcommit=dbcommit,
+                delindex=True,
+            )
+            record = None
+            action = Action.DELETE
+            viaf_record = None
+            online = False
+        else:
+            mef_record, mef_action, viaf_record, online = \
+                record.create_or_update_mef_viaf_record(
+                    dbcommit=dbcommit,
+                    reindex=reindex,
+                    online=online
+                )
+        return record, action, mef_record, mef_action, viaf_record, online
```

Comment:
COSMETIC : missing docstring on return statement (especially here because there are many returned values)

Reply:
If I have time I need to add docstrings everywhere.